lib/jsonschema-tools.js

```diff
@@ -859,10 +896,30 @@ async function materializeAllSchemas(options = {}) {
 
     return _.flatten(await Promise.all(_.map(
         currentSchemasInfo,
-        info => materializeSchemaVersion(path.dirname(info.path), info.schema, options)
+        info => materializeSchemaToPath(path.dirname(info.path), info.schema, options)
     )));
 }
 
+/**
+ * Traverses through all the fields in the schema, including nested properties, and
+ * sets the numeric bounds specified in options when the object type is number.
+ * @param {Object} schema
+ * @param {Array} bounds as specified in options
+ * @return {Object} schema with bounded numeric fields
+ */
+function enforceNumericBounds(schema, bounds) {
+    const schemaCopy = _.cloneDeep(schema);
+    const enforcedMin = bounds[0];
+    const enforcedMax = bounds[1];
+    traverseSchema(schemaCopy, (obj) => {
+        if (['number', 'integer'].includes(obj.type)) {
+            obj.minimum = Math.max(enforcedMin, obj.minimum || -Infinity);
```

Comment:
Hm, I don't think we want to overwrite the `minimum` or `maximum` if they are provided.  We just want to set these to the enforcedMin/Max default values if they aren't set manually by the user.  If I set `maximum` in a current.yaml schema file, I'd expect that to show up in the materialized files, no matter what the default jsonschema-tools settings are.

Reply:
so probably we want: 

```
obj.minimum = obj.minimum || enforcedMin;
```

Reply:
@ottomata but isn't that at odds with the intent of this task? The code as it is will only overwrite the values if the ones the user provided are outside the bounds specified in options.

I think the author of a schema shouldn't be able to set min/max values that are outside those configured by the schema repo administrator.